Design/Design Details for SpectrumInfo.md

```diff
@@ -0,0 +1,71 @@
+There are a number of methods present in [`SpectrumInfo.h`](https://github.com/mantidproject/mantid/blob/43fc616926a32863f37e37f4a107413a0de6dee6/Framework/API/inc/MantidAPI/SpectrumInfo.h) but it may not make sense to expose all of them to be used in Python scripts.
+There are plans to expose [`DetectorInfo`](https://github.com/mantidproject/mantid/blob/ffd49f84bdd2e1bb8f0deeb40727fe775a4974ae/Framework/Geometry/inc/MantidGeometry/Instrument/DetectorInfo.h) and [`ComponentInfo`](https://github.com/mantidproject/mantid/blob/8ec802f56c5db2261a0f9502f30f67fe42530d62/Framework/Geometry/inc/MantidGeometry/Instrument/ComponentInfo.h) as well and so it is important that only the required methods from each class are exposed. As such it was decided that this document should be made so that decisions could be taken about which methods are the most relevant for users. Please see below for a list of the methods that are suggested for exposing to Python.
+
+## Suggested Methods
+
+Method Signature in SpectrumInfo.cpp | Method Usage in Python
+--------------------------------|--------------------------------------
+`size_t size() const` | `int len(SpectrumInfo obj)`
+`size_t size() const` | `int size()`
+`bool isMonitor(const size_t index) const` | `bool isMonitor(int index)`
+`bool isMasked(const size_t index) const` | `bool isMasked(int index)`
+`void setMasked(const size_t index, bool masked)` | `void setMasked(int index, bool masked)`
+`double twoTheta(const size_t index) const` | `double twoTheta(int index)`
+`double signedTwoTheta(const size_t index) const` | `double signedTwoTheta(int index)`
+`double l1() const` | `double l1()`
+`double l2(const size_t index) const` | `double l2(int index)`
+`bool hasDetectors(const size_t index) const` | `bool hasDetectors(int index)`
+`bool hasUniqueDetector(const size_t index) const` | `bool hasUniqueDetector(int index)`
+`Kernel::V3D position(const size_t index) const` | `V3D position(int index)`
+`Kernel::V3D sourcePosition() const` | `V3D sourcePosition()`
+`Kernel::V3D samplePosition() const` | `V3D samplePosition()`
+-- | `list<SpectrumDefinition> getAllSpectrumDefinitions()`
+
+### Further Information
+As can be seen from the above table, the majority of the exported methods retain their name and arguments on the Python side.
+It was also decided that any methods involving `position` or `rotation` would need to have their documentation updated such that the prefix "absolute" is used when referring to those methods. The purpose of this being so that we can distinguish between `Instrument 1.0` and `Instrument 2.0`.
+
+The last change is with regards to the method `getAllSpectrumDefinitions()`. This method is not actually implemented in the C++ `SpectrumInfo` class but it is exported as it was thought that a list of all the `SpectrumDefinition`s would be more natural to users.
```

Comment:
Why is this required? It is inconsistent with everything else in the design and will just be a cumbersome leftover if we were to introduce proper iteration.

Reply:
Oh that is a good point - I'll remove that method from the design document and from the implementation. Do you think we would want to add in iteration like we did for `DetectorInfo` [here](https://github.com/mantidproject/mantid/pull/23226)?

Reply:
Sounds good. I'll make an issue for it.

Reply:
Consider doing it in the same PR as for `DetectorInfo`. There is a chance that you can share a lot of code by templating the iterator.

Reply:
Ah yes maybe. I think the `SpectrumInfo` PR has not yet been merged into master so I probably won't be able to reuse the code until both PRs are in master...